sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2296,6 +2296,14 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
     ZeDesc.flags = 0;
     ZeDesc.ordinal = 0;
 
+    if (Size > Device->ZeDeviceProperties.maxMemAllocSize) {
+      // Tell Level-Zero to accept Size > maxMemAllocSize
+      ze_relaxed_allocation_limits_exp_desc_t RelaxedDesc = {};
```

Comment:
Please put this outside the if, so it is within the scope of the zeMemAllocDevice call in line 2308.

Reply:
Thanks!